.github/workflows/release_prep.yml

```diff
@@ -0,0 +1,44 @@
+name: Release Preparation
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  prepare-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: main
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+
+      - name: Install Changie
+        run: go install github.com/miniscruff/changie@latest
+
+      - name: Batch and Merge Changelog
+        run: |
+          changie batch auto
+          changie merge
+
+      - name: Commit Changelog Changes
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git add .
+          git commit -m "Merge changelog entries for release" || echo "No changes to commit"
+          git push origin main
+
+      - name: Create and Push Release Tag
+        run: |
+          VERSION="v$(changie version)"
```

Comment:
how is this like going to work? You don't have any variable named `changie version`

Reply:
it's literally running the changie command with the version option which returns the current version number

Reply:
It's literally a hallucinated command:
```bash
Usage:
  changie [command]

Available Commands:
  batch       Batch unreleased changes into a single changelog
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  init        Initialize a new changie skeleton
  latest      Echos the latest release version number
  merge       Merge all versions into one changelog
  new         Create a new change file
  next        Next echos the next version based on semantic versioning

Flags:
  -h, --help      help for changie
  -v, --version   version for changie 
```

Reply:
Should be changie latest